res_users_password_security/README.rst

```diff
@@ -0,0 +1,94 @@
+.. image:: https://img.shields.io/badge/license-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+==================
+Password  Security
+==================
+
+This module allows admin to set company-level password security requirements
+and enforces them on the user.
+
+It contains features such as
+
+* Password expiration days
+* Password length requirement
+* Password minimum number of lowercase letters
+* Password minimum number of uppercase letters
+* Password minimum number of numbers
+* Password minimum number of special characters
```

Comment:
I gave it a valiant effort, but my strategy seems to be a bust. These `pbkdf2_*` hashes return a different value each time, similar to encryption. This defeats the normal strategy of saving and referencing the pass hashes directly for use as a history. This just blew my limited knowledge of cryptography out of the water - crazy smart people with their crazy math.

I'm open to ideas if you have them. The last commit removes my attempt if you want to see what I had.

```
>>> pbkdf2_sha512.encrypt("password")
'$pbkdf2-sha512$25000$4ByDsPa.lxJCaE2JkVLK.Q$dVDsbr49NHfKEQWYdjunuqbz2a2MGQmj0rfF3Bh519StAUouvCyYag50EL7kjBckyArqDRBdrwDHkeWeHva5DA'
>>> pbkdf2_sha512.encrypt("password")
'$pbkdf2-sha512$25000$RYixlnKOMab0vjemtNa6lw$d/8zZUv4Ttjfj31/LrXulIk8EuyPPptGpwWllm8jEtU0Ehoe7UfjPmzdn8KwpM8o6eTShCFjn7TvfFzSpf.J3Q'
>>> pbkdf2_sha512.encrypt("password")
'$pbkdf2-sha512$25000$AQBgLEVoLSVEKIUwZuzd.w$BJkX.U1YJTO5Q.g5I/1kh4guNF0qG/1/cYJ8amhFB0JSRT65oPtuJJtwGvNNUGOcdElKuNsXyyitAhcWZdK6Ng'
```